examples/lessons/14/app.js

```diff
@@ -0,0 +1,291 @@
+uniform sampler2D uSampler;
+
+void main(void) {
+    vec3 lightWeighting;
+    if (!uUseLighting) {
+        lightWeighting = vec3(1.0, 1.0, 1.0);
+    } else {
+        vec3 lightDirection = normalize(uPointLightingLocation - vPosition.xyz);
+        vec3 normal = normalize(vTransformedNormal.xyz);
+        float specularLightWeighting = 0.0;
+        if (uShowSpecularHighlights) {
+            vec3 eyeDirection = normalize(-vPosition.xyz);
+            vec3 reflectionDirection = reflect(-lightDirection, normal);
+            specularLightWeighting = pow(max(dot(reflectionDirection, eyeDirection), 0.0), uMaterialShininess);
+        }
+        float diffuseLightWeighting = max(dot(normal, lightDirection), 0.0);
+        lightWeighting = uAmbientColor
+            + uPointLightingSpecularColor * specularLightWeighting
+            + uPointLightingDiffuseColor * diffuseLightWeighting;
+    }
+    vec4 fragmentColor;
+    if (uUseTextures) {
+        fragmentColor = texture2D(uSampler, vec2(vTextureCoord.s, vTextureCoord.t));
+    } else {
+        fragmentColor = vec4(1.0, 1.0, 1.0, 1.0);
+    }
+    gl_FragColor = vec4(fragmentColor.rgb * lightWeighting, fragmentColor.a);
+}
+`;
+
+function getTeapotUniforms() {
+  return {
+    uMaterialShininess: 20.0,
+    uShowSpecularHighlights: true,
+    uUseLighting: true,
+    uUseTextures: true
+  };
+}
+
+function getLightUniforms() {
+  return {
+    uAmbientColor: [0.2, 0.2, 0.2],
+    uPointLightingLocation: [-10.0, 4.0, -20.0],
+    uPointLightingSpecularColor: [0.8, 0.8, 0.8],
+    uPointLightingDiffuseColor: [0.8, 0.8, 0.8]
+  };
+}
+const animationLoop = new AnimationLoop({
+  onInitialize: ({canvas, gl}) => {
+
+    setParameters(gl, {
+      clearColor: [0, 0, 0, 1],
+      clearDepth: 1,
+      depthTest: true,
+      [GL.UNPACK_FLIP_Y_WEBGL]: true
+    });
+
+    return loadFiles({
+      urls: ['Teapot.json']
+    }).then(files => {
+      return loadTextures(gl, {
+        urls: ['arroway.de_metal+structure+06_d100_flat.jpg', 'earth.jpg'],
+        parameters: [{
+          [gl.TEXTURE_MAG_FILTER]: gl.LINEAR,
+          [gl.TEXTURE_MIN_FILTER]: gl.LINEAR_MIPMAP_NEAREST,
+          [gl.TEXTURE_WRAP_S]: gl.REPEAT,
+          [gl.TEXTURE_WRAP_T]: gl.REPEAT,
+          mipmap: true
+        }, {
+          [gl.TEXTURE_MAG_FILTER]: gl.LINEAR,
+          [gl.TEXTURE_MIN_FILTER]: gl.LINEAR_MIPMAP_NEAREST,
+          [gl.TEXTURE_WRAP_S]: gl.REPEAT,
+          [gl.TEXTURE_WRAP_T]: gl.REPEAT,
+          mipmap: true
+        }]
+      })
+      .then(textures => {
+        const galvanizedTexture = textures[0];
+        const earthTexture = textures[1];
+        const teapotJson = JSON.parse(files[0]);
+
+        const fragmentLightingProgram = new Program(gl, {
+          fs: FRAGMENT_LIGHTING_FRAGMENT_SHADER,
+          vs: FRAGMENT_LIGHTING_VERTEX_SHADER,
+        });
+
+        const teapot = new Model(gl, {
+          id: 'teapot-model',
+          program: fragmentLightingProgram,
```

Comment:
Nit: Just pass `fs` and `vs` directly to Model, it will create the program for you.